File: app/src/main/java/com/tesonet/example/android_party/utils/Preferences.java
package com.tesonet.example.android_party.utils;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.tesonet.example.android_party.model.ExListItem;

import java.lang.reflect.Type;
import java.util.ArrayList;

/**
 * Created by Vilius on 2018-03-07.
 */

public class Preferences {
    private static final String PREFERENCE_FILE = "ExampleAppPreferences";

    private static final String PROPERTY_TOKEN = "user_token";
    private static final String PROPERTY_EX_LIST = "ex_list";

    private Preferences() {}

    private static SharedPreferences getSharedPreferences(Context context) {
        return context.getSharedPreferences(PREFERENCE_FILE, Context.MODE_PRIVATE);
    }

    public static void storeTokenValue(Context context, String value) {
        final SharedPreferences prefs = getSharedPreferences(context);
        SharedPreferences.Editor editor = prefs.edit();
        editor.putString(PROPERTY_TOKEN, value);
        editor.apply();
    }

    public static String getTokenValue(Context context) {
        final SharedPreferences prefs = getSharedPreferences(context);
        return prefs.getString(PROPERTY_TOKEN, "");
    }

    public static void storeExList(Context context, ArrayList<ExListItem> values) {
        final SharedPreferences prefs = getSharedPreferences(context);
        SharedPreferences.Editor editor = prefs.edit();
        Gson gson = new Gson();
        String json = gson.toJson(values);
        editor.putString(PROPERTY_EX_LIST, json);
        editor.apply();
    }

    public static ArrayList<ExListItem> getExList(Context context) {
        final SharedPreferences prefs = getSharedPreferences(context);
        String json = prefs.getString(PROPERTY_EX_LIST, null);
        Type listType = new TypeToken<ArrayList<ExListItem>>(){}.getType();
        return new Gson().fromJson(json, listType);
    }
}

File: app/src/main/java/com/alex/tesonettesttask/logic/utils/Preferences.kt
package com.alex.tesonettesttask.logic.utils

import android.content.Context
import android.content.SharedPreferences
import com.alex.tesonettesttask.TesonetApplication

object Preferences {

    private const val APP_SHARED_PREFS = "tesonet.prefs"
    private const val AUTH_TOKEN = "AUTH_TOKEN"
    private val mAppSharedPrefs: SharedPreferences

    init {
        this.mAppSharedPrefs = TesonetApplication.instance.getSharedPreferences(APP_SHARED_PREFS, Context.MODE_PRIVATE)
    }

    fun getAuthToken(): String {
        return mAppSharedPrefs.getString(AUTH_TOKEN, "")
    }

    fun setAuthToken(authToken: String) {
        mAppSharedPrefs.edit().putString(AUTH_TOKEN, authToken).apply()
    }

    fun removeToken() {
        mAppSharedPrefs.edit().remove(AUTH_TOKEN).apply()
    }


}
